### PR TITLE
feat(core,platform): date & datetime picker - ability to process input value on blur

### LINKED ISSUE #7525: Date time component: add config option to parse input text on focus leave (instead of keystroke)
#### Is this a bug, enhancement, or feature request?
enhancement

#### Briefly describe your proposal.

Screenshot 2022-01-10 at 14 15 00

Editing the formatted date time string shown `<fd-datetime-picker>` component's input field triggers function `DatetimeAdapter.parse(...)` on "keystroke".

I'd like to request to be able to configure so `DatetimeAdapter.parse(...)` is only triggered on "focus leave"

Why?
When `DatetimeAdapter.parse(...)` returns a valid date, then `DatetimeAdapter.format(...)` is called and replaces current text in the input element. 

Example:
Initial text: `15.01.2022 13:00`
My target date: `17.01.2022 13:00`

Steps from initial to target:

1. `15.01.2022 13:00` (parsed date: 2022-01-15T13:00:00.000Z)
2. `1.01.2022 13:00` (parsed date: 2022-01-01T13:00:00.000Z)
3. `DatetimeAdapter.format()` sets new text to `01.01.2022 13:00` <= having this happend is undesirable

Preferred steps:
1. `15.01.2022 13:00` (no parsing)
2. `1.01.2022 13:00` (no parsing)
3. `17.01.2022 13:00` (no parsing)
4. Focus cursor to next field (NOW DO PARSING)
5. `17.01.2022 13:00` (parsed date: 2022-01-17T13:00:00.000Z)

Alternative solution: I looked into modifying the parse function, but this is not an option. "1.01.2022 13:00" is and should be parse to a valid date. Flagging "1.01.2022 13:00" as an invalid date format does not seem correct to me. I would prefer to configure when the input text is parsed. 

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
Fundamental Library: 0.25.X
Angular: 12.2.X

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
## Related Issue(s)

closes #7525 

## Description
Currently input updates it's value and validity as user types. Adding an ability to apply changes on leaving the input or by hitting enter.

Functionality has been added to:
- core-date-picker
- core-datetime-picker
- platform-date-picker
- core-datetime-picker

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Need to remove this text from `date-picker-docs.component.html`: ```The value for the input only changes the model on blur or an enter/return keystroke.```

File: apps/docs/src/app/platform/component-docs/platform-forms/platform-datetime-picker/platform-datetime-picker-examples/platform-datetime-picker-update-on-blur-example.component.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    selector: 'fdp-platform-datetime-picker-update-on-blur-example',
+    template: `<label for="update-on-blur">Datetime picker:</label>
+        <fdp-datetime-picker
+            name="update-on-blur"
+            placeholder="MM/dd/YYYY, hh:mm"
+            [(ngModel)]="date"
+            [processInputOnBlur]="true"
+        ></fdp-datetime-picker>
+        <br />
+        <div>Selected Date: {{ date?.toDateString() || 'null' }}</div> `,
```
Comment (mikerodonnell89) [inline apps/docs/src/app/platform/component-docs/platform-forms/platform-datetime-picker/platform-datetime-picker-examples/platform-datetime-picker-update-on-blur-example.component.ts:21]: Should show the full date + time here

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Left a couple comments

**Comment by dmitry-stepanenko** [comment]
@mikerodonnell89 done

File: libs/core/src/lib/date-picker/date-picker.component.ts [hunk 2]
```diff
@@ -277,6 +277,14 @@ export class DatePickerComponent<D> implements OnInit, OnDestroy, AfterViewInit,
     @Input()
     inline = true;
 
+    /**
+     * Whether to recalculate value from the input as user types or on blur.
+     * By default, updates the value as user types.
+     * @default false
+     */
+    @Input()
+    processInputOnBlur = false;
```
Comment (platon-rov) [inline libs/core/src/lib/date-picker/date-picker.component.ts:286]: Let's use `coerceBoolenProperty` from sdk here.

File: libs/core/src/lib/datetime-picker/datetime-picker.component.ts [hunk 3]
```diff
@@ -259,6 +259,14 @@ export class DatetimePickerComponent<D>
     @Input()
     showFooter = true;
 
+    /**
+     * Whether to recalculate value from the input as user types or on blur.
+     * By default, updates the value as user types.
+     * @default false
+     */
+    @Input()
+    processInputOnBlur = false;
```
Comment (platon-rov) [inline libs/core/src/lib/datetime-picker/datetime-picker.component.ts:268]: coercion.

**Review by platon-rov — APPROVED** [review]

**Comment by dmitry-stepanenko** [comment]
@platon-rov done

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request merged**